articleview.cc

```diff
@@ -616,36 +595,19 @@ void ArticleView::loadFinished( bool )
     {
       QString originalAnchor = anchor.mid( n + 1 );
 
-      QRegExp rx;
-      rx.setMinimal( true );
-      rx.setPattern( anchor.left( 34 ) + "[0-9a-f]*_" + originalAnchor );
-
-      QWebElementCollection coll = ui.definition->page()->mainFrame()->findAllElements( "a[name]" );
-      coll += ui.definition->page()->mainFrame()->findAllElements( "a[id]" );
-
-      for( QWebElementCollection::iterator it = coll.begin(); it != coll.end(); ++it )
-      {
-        QString name = (*it).attribute( "name" );
-        QString id = (*it).attribute( "id" );
-        if( ( !name.isEmpty() && rx.indexIn( name ) >= 0 )
-            || ( !id.isEmpty() && rx.indexIn( id ) >= 0 ))
-        {
-          // Anchor found, jump to it
-
-          url.clear();
-          url.setFragment( rx.cap( 0 ) );
-          ui.definition->page()->mainFrame()->evaluateJavaScript(
-             QString( "window.location.hash = \"%1\"" ).arg( QString::fromUtf8( url.toEncoded() ) ) );
-
-          break;
-        }
-      }
+      int end = originalAnchor.indexOf('_');
+      QString hash=originalAnchor.left(end);
+      url.clear();
+      url.setFragment(hash);
+      ui.definition->page()->runJavaScript(
+          QString("window.location.hash = \"%1\"").arg(QString::fromUtf8(url.toEncoded())));
```

Comment:
I am not using MDict dictionaries myself. So hopefully MDict users can clarify this code and change to me:
1. Doesn't this simplification break some MDict anchor jumps?
2. The current code in GoldenDict master looks like a risky workaround that could break non-MDict anchor jumps if they happen to have an underscore at the index 33:
```cpp
    int n = anchor.indexOf( '_' );
    if( n == 33 )
      // MDict pattern: ("g" + dictionary ID (33 chars total))_(articleID(quint64, hex))_(original anchor)
      n = anchor.indexOf( '_', n + 1 );
    else
      n = 0;
```
Any idea how to make this workaround more robust and maybe simplify the code to falicitate clean porting to Qt WebEngine?